Shaders/UIMask.fx

```diff
@@ -1,189 +1,266 @@
+	copies of the Software, and to permit persons to whom the Software is
+	furnished to do so, subject to the following conditions:
+
+	The above copyright notice and this permission notice shall be included in all
+	copies or substantial portions of the Software.
+
+	THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+	IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+	FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+	AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+	LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+	OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+	SOFTWARE.
 */
 
+//#region Preprocessor
+
 #include "ReShade.fxh"
+#include "ReShadeUI.fxh"
 
 #ifndef UIMASK_MULTICHANNEL
-    #define UIMASK_MULTICHANNEL 0
+	#define UIMASK_MULTICHANNEL 0
 #endif
 
 #ifndef UIMASK_TOGGLEKEY_RED
-    #define UIMASK_TOGGLEKEY_RED 0x67 //Numpad 7
+	#define UIMASK_TOGGLEKEY_RED 0x67 //Numpad 7
 #endif
 
 #ifndef UIMASK_TOGGLEKEY_GREEN
-    #define UIMASK_TOGGLEKEY_GREEN 0x68 //Numpad 8
+	#define UIMASK_TOGGLEKEY_GREEN 0x68 //Numpad 8
 #endif
 
 #ifndef UIMASK_TOGGLEKEY_BLUE
-    #define UIMASK_TOGGLEKEY_BLUE 0x69 //Numpad 9
+	#define UIMASK_TOGGLEKEY_BLUE 0x69 //Numpad 9
 #endif
 
 #if !UIMASK_MULTICHANNEL
-    #define TEXFORMAT R8
+	#define TEXFORMAT R8
 #else
-    #define TEXFORMAT RGBA8
+	#define TEXFORMAT RGBA8
 #endif
 
-#include "ReShadeUI.fxh"
+//#endregion
+
+namespace UIMask
+{
 
-uniform float fMask_Intensity < __UNIFORM_SLIDER_FLOAT1
-    ui_label = "Mask Intensity";
-    ui_tooltip = "How much to mask effects to the original image.";
-    ui_min = 0.0;
-    ui_max = 1.0;
-    ui_step = 0.001;
+//#region Uniforms
+
+uniform float fMask_Intensity
+<
+	__UNIFORM_SLIDER_FLOAT1
+
+	ui_label = "Mask Intensity";
+	ui_tooltip =
+		"How much to mask effects to the original image.\n"
+		"\nDefault: 1.0";
+	ui_min = 0.0;
+	ui_max = 1.0;
+	ui_step = 0.001;
 > = 1.0;
 
 uniform bool bDisplayMask <
-    ui_label = "Display Mask";
-    ui_tooltip = 
-        "Display the mask texture.\n"
-        "Useful for testing multiple channels or simply the mask itself.";
+	ui_label = "Display Mask";
+	ui_tooltip =
+		"Display the mask texture.\n"
+		"Useful for testing multiple channels or simply the mask itself.\n"
+		"\nDefault: Off";
 > = false;
 
 #if UIMASK_MULTICHANNEL
-uniform bool toggle_red <source="key"; keycode=UIMASK_TOGGLEKEY_RED; toggle=true;>;
-uniform bool toggle_green <source="key"; keycode=UIMASK_TOGGLEKEY_GREEN; toggle=true;>;
-uniform bool toggle_blue <source="key"; keycode=UIMASK_TOGGLEKEY_BLUE; toggle=true;>;
-#endif
 
-texture tUIMask_Backup { Width = BUFFER_WIDTH; Height = BUFFER_HEIGHT; };
-texture tUIMask_Mask <source="UIMask.png";> { Width = BUFFER_WIDTH; Height = BUFFER_HEIGHT; Format=TEXFORMAT; };
+uniform bool ToggleRed
```

Comment:
Better leave these with their original name, so not to break presets. That also means not using a namespace.

Reply:
Do presets take texture names in account?
I didn't know that. Sure, will do.

Reply:
They don't, I was talking about the uniform variables. The texture names can change =).

Reply:
Oh sure, but do uniforms with source attributes matter?

Reply:
They don't either .... It was late when I looked at this yesterday, please ignore =P